projects/openapi-cli/src/captures/system-proxy.ts

```diff
@@ -0,0 +1,90 @@
+import { exec } from 'child_process';
+import url from 'url';
+import { createCommandFeedback } from '../commands/reporters/feedback';
+
+const platform: 'mac' | 'windows' | 'linux' =
+  process.platform === 'win32'
+    ? 'windows'
+    : process.platform === 'darwin'
+    ? 'mac'
+    : 'linux';
+
+export class SystemProxy {
+  constructor(
+    private proxyUrl: string,
+    private feedback: Awaited<ReturnType<typeof createCommandFeedback>>
+  ) {}
+  private stopCommand: (() => Promise<void>) | undefined = undefined;
+
+  async start(device: string | undefined) {
+    const { port } = url.parse(this.proxyUrl);
+
+    if (platform === 'mac') {
```

Comment:
right now the magic only works on Mac. There are clear paths to turn it on w/ windows, but that's something we should validate demand for first via mixpanel